### PR TITLE
Connection improvements and bug fixes

### PR DESCRIPTION
**Pull request by oliveira**
Fixes:
* Connection data now is updating after changes on form.
* Added connection validation to go to main component.
* Select table fixed by @gabrielboliveira 
* Added default data(just not filling fields)
* Now if connection is ok also table is valid, load table data.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -288,12 +303,13 @@ export default {
                 })
 
                 this.setDatabaseActive(this.connection.database)
+                if (this.databases.includes(this.connection.database)) this.loadTables(this.connection.database)
```
Comment (gabrielboliveira) [inline src/App.vue]:
```js
if (this.databases.includes(this.connection.database)) {
     this.loadTables(this.connection.database)
}
```

File: src/App.vue [hunk 2]
```diff
@@ -231,48 +238,61 @@ export default {
...
-                    ranAt: new Date(),
-                    sql: data.sql.replace(/\?/g, () => {
-                        let value = data.bindings[index++]
+        // TODO: think a better way to test connection
+        testConnection() {
+            this.setDefaultValues()
+            this.knex = this.getKnex()
+
+            return this.loadDatabases()
+                            .then(() => {
```
Comment (wesleyhf) [inline src/App.vue:260]: use only one tab

File: src/App.vue [hunk 3]
```diff
@@ -231,48 +238,61 @@ export default {
...
+                        if (data.method === 'pluck') return
+
+                        let index = 0
+
+                        this.setQueryLog({
+                            ranAt: new Date(),
+                            sql: data.sql.replace(/\?/g, () => {
+                                let value = data.bindings[index++]
+
+                                return _isNumber(value) || _isNull(value) ?
```
Comment (wesleyhf) [inline src/App.vue:285]: put inline

File: src/App.vue [hunk 4]
```diff
@@ -282,18 +302,20 @@ export default {
...
-                })
-
+            return this.knex.raw('show databases').then(databases => {
+                this.databases = databases[0].map(db => db.Database)
                 this.setDatabaseActive(this.connection.database)
+
+                if (this.databases.includes(this.connection.database)) {
+                    this.loadTables(this.connection.database)
+                }
+                return true
```
Comment (wesleyhf) [inline src/App.vue:312]: enter

File: src/App.vue [hunk 5]
```diff
@@ -231,48 +238,61 @@ export default {
...
-            this.knex.on('query', data => {
-                // pluck are the internal queries to get databases, tables and columns
-                if (data.method === 'pluck') return
+        replaceEmpty(obj, def) {
+            for (var prop in obj) {
+                if (obj[prop]) continue;
 
-                let index = 0
+                obj[prop] = def[prop]
+            }
```
Comment (gabrielboliveira) [inline src/App.vue:251]: Please use lodash's [forEach](https://lodash.com/docs/#forEach)

File: src/App.vue [hunk 6]
```diff
@@ -231,48 +238,61 @@ export default {
...
+
+                                return _isNumber(value) || _isNull(value) ?
+                                    value :
+                                    `'${value}'`
+                            }),
+                        })
+                    })
+
+                    this.loadDatabases()
+                    return true
```
Comment (wesleyhf) [inline src/App.vue:293]: enter

**Review by wesleyhf — CHANGES_REQUESTED** [review]

File: src/App.vue [hunk 7]
```diff
@@ -231,48 +238,61 @@ export default {
...
+                        let index = 0
+
+                        this.setQueryLog({
+                            ranAt: new Date(),
+                            sql: data.sql.replace(/\?/g, () => {
+                                let value = data.bindings[index++]
+
+                                return _isNumber(value) || _isNull(value) ?
+                                    value :
+                                    `'${value}'`
```
Comment (gabrielboliveira) [inline src/App.vue:287]:
```js
return _isNumber(value) || _isNull(value)
    ? value
    : `'${value}'`
```

File: src/App.vue [hunk 8]
```diff
@@ -231,48 +238,61 @@ export default {
...
+                            })
+        },
 
-                        return _isNumber(value) || _isNull(value)
-                            ? value
-                            : `'${value}'`
-                    }),
+        connect() {
+            this.testConnection()
+                .then(() => {
```
Comment (gabrielboliveira) [inline src/App.vue:268]:
```js
this.testConnection().then(() => this.parseTestConnection())
```

File: src/app/component/explorer/index.vue [hunk 9]
```diff
@@ -191,6 +191,7 @@ import {
     without as _without,
     clone as _clone,
 } from 'lodash'
+import Vue from 'vue'
```
Comment (gabrielboliveira) [inline src/app/component/explorer/index.vue:194]: You really need this import?

**Review by gabrielboliveira — CHANGES_REQUESTED** [review]

**Pull request merged**